packages/prover/src/web3_provider_inspector.ts

```diff
@@ -0,0 +1,89 @@
+import {Logger} from "@lodestar/logger";
+import {AnyWeb3Provider, Web3ProviderType} from "./interfaces.js";
+
+import web3jsProviderType from "./provider_types/web3_js_provider_type.js";
+import ethersProviderType from "./provider_types/ethers_provider_type.js";
+import eip1193ProviderType from "./provider_types/eip1193_provider_type.js";
+import legacyProviderType from "./provider_types/legacy_provider_type.js";
+
+export class Web3ProviderInspector {
+  protected providerTypes: Web3ProviderType<AnyWeb3Provider>[] = [];
+  logger: Logger;
+
+  protected constructor(opts: {logger: Logger}) {
+    this.logger = opts.logger;
+  }
+
+  static initWithDefault(opts: {logger: Logger}): Web3ProviderInspector {
+    const inspector = new Web3ProviderInspector(opts);
+    inspector.register(web3jsProviderType, {index: 0});
```

Comment:
Looks like indexes could be eluded?

Reply:
Yes from here these can be avoided, these will always append in the last. Added indexes here to for clarity.